prov/shm/src/smr_signal.h

```diff
@@ -48,13 +48,15 @@ static void smr_handle_signal(int signum, siginfo_t *info, void *ucontext)
 		shm_unlink(ep_name->name);
 	}
 
-	/* Register the original signum handler, SIG_DFL or otherwise */
-	ret = sigaction(signum, &old_action, NULL);
-	if (ret)
-		return;
+	if (signum < SIGRTMIN) {
```

Comment:
This can only happen if we call smr_reg_sig_hander() with a signal < SIGRTMIN, so this can just be an assert.

Reply:
I guess we could assert in SHM_INI that SIGBUS < SIGRTMIN && SIGSEGV < SIGRTMIN ...

Reply:
I added the assertion in SHM_INI to make sure the four signals we care about are all less than SIGRTMIN.